source/Missions/Services/Missiles/Handlers/MissileHandler.cs

```diff
@@ -0,0 +1,87 @@
+﻿using Autofac;
+using Common;
+using Common.Messaging;
+using Common.Network;
+using GameInterface.Serialization;
+using LiteNetLib;
+using Missions.Services.Agents.Messages;
+using Missions.Services.Agents.Packets;
+using Missions.Services.Missiles.Message;
+using Missions.Services.Missiles.Patches;
+using Missions.Services.Network;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using TaleWorlds.Core;
+using TaleWorlds.Engine;
+using TaleWorlds.Library;
+using TaleWorlds.MountAndBlade;
+
+namespace Missions.Services.Missiles.Handlers
+{
+    public class MissileHandler
+    {
+        IContainer container;
+        IBinaryPackageFactory binaryPackageFactory;
+        public MissileHandler()
+        {
+            ContainerBuilder builder = new ContainerBuilder();
+            builder.RegisterModule<MissionModule>();
+            container = builder.Build();
+
+            binaryPackageFactory = container.Resolve<IBinaryPackageFactory>();
```

Comment:
This is an anti-pattern, do the following instead
```suggestion
        public MissileHandler(IBinaryPackageFactory binaryPackageFactory)
        {
            this.binaryPackageFactory = binaryPackageFactory;
```